sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -203,7 +194,7 @@ class AccountState(
 
       // update current base fee
       // TODO: should use updated base fee based on consumed gas in the new block
-      stateView.updateBaseFee(BigInteger.valueOf(mod.header.baseFee))
+      stateView.updateBaseFee(mod.header.baseFee)
```

Comment:
We could update this to the new base fee, considering gas consumption of the block that was just applied, see TODO comment.

Reply:
right.
`stateView.baseFee` is used in the validate(tx) method, which is used during tx apply to mempool.
So, if `block1` with baseFee `BF1` is the chain tip, then every tx must be validated against `BF2 = nextBaseFee(block1)`